vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/components/ExperimentalVariablesDialog.java

```diff
@@ -26,25 +28,65 @@
  *
  * @since 1.0.0
  */
-public class AddExperimentalVariablesDialog extends DialogWindow {
+public class ExperimentalVariablesDialog extends DialogWindow {
 
   @Serial
   private static final long serialVersionUID = 5296014328282974007L;
-  private final List<AddExperimentalVariablesDialog.ExperimentalVariableRowLayout> experimentalVariablesLayoutRows = new ArrayList<>();
-  private final List<ComponentEventListener<CancelEvent<AddExperimentalVariablesDialog>>> listenersCancellation = new ArrayList<>();
+  private final List<ExperimentalVariablesDialog.ExperimentalVariableRowLayout> experimentalVariablesLayoutRows = new ArrayList<>();
+  private final List<ComponentEventListener<CancelEvent<ExperimentalVariablesDialog>>> listenersCancellation = new ArrayList<>();
   private final Div dialogueContentLayout = new Div();
   private final Div experimentalVariableRowsContainerLayout = new Div();
   private final Span addExperimentalVariableLayoutRow = new Span();
-  private final List<ComponentEventListener<ConfirmEvent<AddExperimentalVariablesDialog>>> listenersConfirmation = new ArrayList<>();
+  private final List<ComponentEventListener<ConfirmEvent<ExperimentalVariablesDialog>>> listenersConfirmation = new ArrayList<>();
+  private final MODE mode;
```

Comment:
can the mode not be changed?

Reply:
no, you instanciate a new dialog in either mode